src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -308,6 +309,31 @@ class RunSetupPyResult:
     output: Digest  # The state of the chroot after running setup.py.
 
 
+class PythonDistributionSubsystem(Subsystem):
+    """Options for packaging wheels/sdists from a `python_distribution` target."""
+
+    options_scope = "python-distribution"
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--exact-first-party-dependency-requirements",
+            type=bool,
+            default=False,
```

Comment:
I think we should default to `~=`, but curious what you all think.

Note that this is an API change. If this weren't 2.0, we would need to set the default to using `==` and use a deprecation cycle to be able to change the default.

Reply:
I went back to using the prior default of `==`, because imo the default should favor correctness over ergonomics. Users can loosen it if they are okay with the risk of less correctness.

Also, we avoid a deprecation cycle.